src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.cs

```diff
@@ -0,0 +1,459 @@
+            {
+                operationName = pOperation.Property.Name;
+            }
+            else if (operation is IFieldReferenceOperation fOperation)
+            {
+                operationName = fOperation.Field.Name;
+            }
+
+            context.ReportDiagnostic(operation.CreateDiagnostic(SwitchRule(attribute.AttributeType), operationName, attribute.OsPlatformName, attribute.Version.ToString()));
+        }
+
+        private static void AnalyzeInvocationOperation(IOperation operation, OperationAnalysisContext context,
+            List<PlatformAttributeInfo>? parsedTfms, ref PooledDictionary<IOperation, ImmutableArray<PlatformAttributeInfo>> platformSpecificOperations)
+        {
+            using var builder = ArrayBuilder<PlatformAttributeInfo>.GetInstance();
+            AttributeData[]? attributes = null;
+
+            if (operation is IInvocationOperation iOperation)
+            {
+                attributes = FindAllPlatformAttributesApplied(iOperation.TargetMethod.GetAttributes(), iOperation.TargetMethod.ContainingType);
+            }
+            else if (operation is IPropertyReferenceOperation pOperation)
+            {
+                if (!IsWithinConditionalOperation(operation))
+                    attributes = FindAllPlatformAttributesApplied(pOperation.Property.GetAttributes(), pOperation.Property.ContainingType);
+            }
+            else if (operation is IFieldReferenceOperation fOperation)
+            {
+                if (!IsWithinConditionalOperation(operation))
+                    attributes = FindAllPlatformAttributesApplied(fOperation.Field.GetAttributes(), fOperation.Field.ContainingType);
+            }
+
+            if (attributes != null)
+            {
+                foreach (AttributeData attribute in attributes)
+                {
+                    if (PlatformAttributeInfo.TryParseAttributeData(attribute, out PlatformAttributeInfo parsedAttribute))
+                    {
+                        if (!(IsSuppressedByTfm(parsedTfms, parsedAttribute) || IsSuppressedByAttribute(parsedAttribute, context.ContainingSymbol)))
+                        {
+                            builder.Add(parsedAttribute);
+                        }
+                    }
+                }
+            }
+
+            if (builder.Count > 0)
+            {
+                platformSpecificOperations.Add(operation, builder.ToImmutable());
+            }
+        }
+
+        private static bool IsWithinConditionalOperation(IOperation pOperation) =>
+         pOperation.Parent is IBinaryOperation bo &&
+            (bo.OperatorKind == BinaryOperatorKind.Equals ||
+            bo.OperatorKind == BinaryOperatorKind.NotEquals ||
+            bo.OperatorKind == BinaryOperatorKind.GreaterThan ||
+            bo.OperatorKind == BinaryOperatorKind.LessThan ||
+            bo.OperatorKind == BinaryOperatorKind.GreaterThanOrEqual ||
+            bo.OperatorKind == BinaryOperatorKind.LessThanOrEqual);
```

Comment:
We talked about excluding  Equal and NotEqual. not sure if we want to exclude others too (if its numeric property or enum then might be useful)

Reply:
Yeah, I think your list looks good for now. We'll likely have to tune this further.